### ISSUE TITLE
Increasing mtry crashes ranger fit

### ISSUE THREAD

**Issue by tomauer** [imbs-hl/ranger#674]
I've tried this on a few different environments and have had good luck reproducing it. If I increase `mtry` beyond 12 (which is equivalent to the default), it immediately crashes with a segfault.

 *** caught segfault ***
address 0x145ad9842000, cause 'invalid permissions'
malloc(): invalid next size (unsorted)
Aborted (core dumped)

I've attached example data and code.

```
x_unique <- readr::read_csv("~/Downloads/x_unique.csv")
x_unique$zi <- as.factor(as.numeric(x_unique$zi))

prop_detections <- 0.19

rf_occ <- ranger::ranger(
  formula = zi ~ .,
  num.trees = 300,
  mtry = 24,
  importance = "impurity",
  num.threads = 1,
  respect.unordered.factors = "order",
  always.split.variables = c("island", "circular_day_of_year_sin", "circular_day_of_year_cos", "has_intertidal", "has_shoreline", "has_evi"),
  probability = TRUE,
  replace = TRUE,
  sample.fraction = c(prop_detections, prop_detections),
  data = x_unique
)
```

Any advice for resolving this so that I can explore tuning this model would be greatly appreciated.

[Archive.zip](https://github.com/imbs-hl/ranger/files/11774654/Archive.zip)

**Comment by tomauer** [comment]
One sessionInfo()

```
R version 4.2.2 (2022-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ranger_0.15.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.10     lattice_0.20-45 fansi_1.0.4     tzdb_0.4.0      utf8_1.2.3      grid_4.2.2      R6_2.5.1        lifecycle_1.0.3 magrittr_2.0.3 
[10] pillar_1.9.0    rlang_1.1.1     cli_3.6.1       rstudioapi_0.14 Matrix_1.5-1    vctrs_0.6.3     ellipsis_0.3.2  tools_4.2.2     readr_2.1.4    
[19] glue_1.6.2      hms_1.1.2       compiler_4.2.2  pkgconfig_2.0.3 tibble_3.2.1 
```

I'm also on machines with a lot memory. Mac M2 with 96GB and Linux cluster with 128GB.

**Comment by mnwright** [comment]
Thanks, this was a bug related to `always.split.variables`, which is fixed with #678.

**Comment by tomauer** [comment]
Works great. Thank you!
